streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -274,7 +274,11 @@ void shutdown(final boolean clean) {
         standby.close(clean);
 
         // remove the changelog partitions from restore consumer
-        restoreConsumer.unsubscribe();
+        try {
+            restoreConsumer.unsubscribe();
+        } catch (final RuntimeException fatalException) {
```

Comment:
Why this change? AFAIK, `unsubscribe()` will never throw.

Reply:
It throws in the test since the consumer was never opened.